src/liboslexec/runtimeoptimize.cpp

```diff
@@ -3035,7 +3041,7 @@ RuntimeOptimizer::run ()
     // Optimize each layer again, from last to first (because some
     // optimizations are only apparent when the subsequent shaders have
     // been simplified).
-    for (int layer = nlayers-2;  layer >= 0;  --layer) {
+    for (int layer = nlayers-1;  layer >= 0;  --layer) {
```

Comment:
TL;DR: it was fine before, but this change gives us a reason why we would want to revisit the last layer a second time, whereas before there was no reason.